OSDOCS-5387: add kubeconfig management (#57834)
openshift/openshift-docs: modules/microshift-kubeconfig-local-access.adoc

// Module included in the following assemblies:
//
// * microshift/microshift_configuring/microshift-kubeconfig.adoc

:_content-type: CONCEPT
[id="microshift-kubeconfig-local-access_{context}"]
= Local access kubeconfig file

> The local access `kubeconfig` file generates in the `/var/lib/microshift/resources/kubeadmin/kubeconfig` directory. This `kubeconfig` file provides access to the API server using `localhost`. Choose this file when you are connecting the cluster locally.

Reviewer: The path given is a file, not a directory.

```suggestion
The local access `kubeconfig` file is written to`/var/lib/microshift/resources/kubeadmin/kubeconfig`. This `kubeconfig` file provides access to the API server using `localhost`. Choose this file when you are connecting the cluster locally.
```

.Example contents of `kubeconfig` for local access
[source, YAML]
----
clusters:
- cluster:
    certificate-authority-data: <base64 CA>
    server: https://localhost:6443
----

> When the cluster running on the {product-title} host connects to the API server locally, the certificate authority (CA) used for validation lists `localhost` as the only Subject Alternative Name (SAN). This is true whether a `config.yaml` exists or built-in values are being used. The main IP address of the host is the default value and cannot be changed. The CA of the local access `kubeconfig` file cannot be used outside of the {product-title} host.

Reviewer: Trying to simplify and remove some implementation details to make the point a bit clearer.

```suggestion
The localhost kubeconfig file can only be used from a client connecting to the API server from the same host. The certificates in the file do not work for remote connections.
```

// Module included in the following assemblies:
//
// * microshift/microshift_networking/microshift-kubeconfig.adoc

:_content-type: CONCEPT
[id="local-access_{context}"]
= Local access kubeconfig file

> The local access `kubeconfig` file is generates in the `/var/lib/microshift/resources/kubeadmin/kubeconfig` directory. This configuration file allows access to the local API server using `localhost` and the internal CA of `localhost`. The CA of the local access `kubeconfig` file cannot be used outside of the {product-title} host.

Reviewer: ```suggestion
The local access `kubeconfig` file is generated in the `/var/lib/microshift/resources/kubeadmin` directory. This configuration file allows access to the local API server using `localhost` and the internal certificate authority (CA) of `localhost`. The CA of the local access `kubeconfig` file cannot be used outside of the {product-title} host.
```

@pacevedom is that right, the local kubeconfig file can't be copied to another host? If so, we should be more explicit about that in the statement here for someone who does not know what a "CA" is.

Reviewer 2: Yes, this file is only valid for localhost access now, it does not work if you copy it to a different host.
We can include that each access type (local, remote) is served a different certificate, each signed by a different CA and that is what the different kubeconfigs are for.

I would also remove the `local` in `allows access to the local API server`.

> When the cluster within the {product-title} host connects to the API server locally, a certificate is used for validation. The certificate lists `localhost` as the only Subject Alternative Name (SAN). You must use `localhost` to connect to the local API server. The `kubeconfig` file must use the same certificate authority (CA) that signed the certificate.

Reviewer: This paragraph starts out with "the cluster" as the actor and then gives the user instructions ("you must"). Maybe we should split it up to explain how the cluster does authentication internally, then tell the user to use a specific kubeconfig when trying to access the API server from the same host.

> Local access `kubeconfig` file values are taken from {product-title} built-in default values or a `config.yaml` file. You cannot add additional values, such as SANs, to the local access `kubeconfig` file. To add SANs to your deployment, you must use the remote access `kubeconfig` file or add them to a `config.yaml` file.

Reviewer: I don't understand what the first sentence here means. What values are taking from `config.yaml`?

Maybe instead of focusing on what users can and cannot do with specific files, this would be clearer if we showed them how to do things we think they are going to want to do?

* Access the API server from the host running MicroShift
* Access the API server from another host using the IP of the MicroShift host
* Access the API server from another host using the hostname of the MicroShift host

What else?

Author: @dhellmann Would this be a good spot to call the other access modules we were discussing in https://issues.redhat.com/browse/OCPBUGS-10029?

Author: @pacevedom What do you think?
